src/main/java/com/marklogic/appdeployer/command/groups/DeployGroupsCommand.java

```diff
@@ -35,11 +48,19 @@ protected ResourceManager getResourceManager(CommandContext context) {
     protected void afterResourceSaved(ResourceManager mgr, CommandContext context, File resourceFile,
             SaveReceipt receipt) {
         String payload = receipt.getPayload();
-        if (payload != null && payload.contains("cache-size") && context.getAdminManager() != null) {
-            if (logger.isDebugEnabled()) {
-                logger.info("Group payload contains cache-size parameter, so waiting for ML to restart");
-            }
-            context.getAdminManager().waitForRestart();
+        if (payload != null) {
+        	if (payload.contains("cache-size") && context.getAdminManager() != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.info("Group payload contains cache-size parameter, so waiting for ML to restart");
+                }
+                context.getAdminManager().waitForRestart();
+        	}        	
+			// When new groups are created, an Admin server is automatically created in that group.
+			// However, the Admin server's rewrite property is empty - causing problems with reading the timestamp
+        	Fragment xml = new Fragment(payload);
+        	String groupName = xml.getElementValue("/m:group-properties/m:group-name");
```

Comment:
Use PayloadParser here as a safe way to get the value, no matter if the payload is XML or JSON (this code won't work on a JSON payload)